### ISSUE TITLE
Conditions for determining a colony state

### ISSUE THREAD

**Issue by TheAtom** [Idhrendur/paradoxGameConverters#405]
This concerns "colony" vs. "state" status of a V2 state. Presently, a converted state is turned into a colony if it meets all of the below:
1) it is not on the same continent as the capital,
2) it is not connected by land,
3) it was colonized or "infidel-conquered".
It is converted into a state otherwise.

"Infidel-conquered" means that any (EU4) province in the state satisfies:
a) the province's original religion is considered heathen by the current owner's religion, **AND**
b) the province's original culture is different from its current culture.

This means that the state will not be turned into a colony if its provinces' cultures remain original. If the conqueror converts one of those provinces to other culture (i.e. it's own), the state will become a colony.

Additionaly, the code doesn't seem to care about the original "territory" vs. "state" status of the EU4 states that the provinces belong to.

Proposed solution: replace b) with "b') the province's original culture is not an accepted culture by the province's current owner".

Note that normally in order to elevate a culture to an accepted status, some state of this culture must be first elevated to "state" from "territory".

It would also be nice to somehow force all overseas, unconnected territories to be converted to colonies no matter what. The problem is obviously that EU4 states may not match V2 states.

**Comment by kratostatic** [comment]
Agreed with your main proposal.  I'm not convinced that forcing all overseas unconnected territories to be colonial works all that well in practice - even aside from the problem you mention things get very awkward as the Bosphoros doesn't count as a land connection, so Turkish/Byzantine Syria ends up being overseas.

**Comment by TheAtom** [comment]
The converter is supposed to use V2 data for land connections, and Bosphoros counts as a land connection (provinces 860 and 861 in adjacencies.csv). It uses EU4 continent data for "same continent check", though.
 I write "supposed" because while checking all this, I stumbled upon this: 
https://github.com/Idhrendur/paradoxGameConverters/blob/c3a4898094bf7c5e5a262d65345190b46f9f779d/EU4toV2/Source/Mappers/AdjacencyMapper.cpp#L168-L172
And this function (well, a wrapper around it) is used in the code. Even though adjacencies are read from the adjacency binary file correctly, the converter never uses it. This must be the cause of a number of bugs.

On the subject of colonies, if all straits in V2 count as a land connection, what are some other awkward scenarios?

**Comment by kratostatic** [comment]
Sorry, coming back to this after some time.  Yeah that adjacency mapper cannot be working as intended!

Another awkward scenario could be a cross Mediterranean empire, especially one not at Gibraltar and Tangier.  Quite often in CK2 Italian states will conquer Tunis or France Algiers for example, and I rather suspect that if it has been part of the country for potentially 1000 years or more by this point it should be considered colonial.  

Yemen etc conquering in east Africa could also be awkward - need to confirm if Omani Zanzibar is colonial or not in vanilla as that's a pretty close analogy.  

Is there a strait across the Persian gulf in Vicky 2?  Because there is in EU4 and countries often conquer land on either side of it.
 
A Haiti might conquer chunks of South America.  

Probably a rare case but probably the worst - one of the Malay states might change its capital to Australia (especially if played by a player) in which case its _original_ land would be colonial!

**Comment by Idhrendur** [comment]
Given modern EU4, it might be best to have EU4 states (as opposed to territories) become full states.
